RetailCoder.VBE/Inspections/ObjectVariableNotSetInspection.cs

```diff
@@ -2,62 +2,45 @@
         {
-            var interestingDeclarations =
-                State.AllUserDeclarations.Where(item =>
-                        !item.IsSelfAssigned &&
-                        !item.IsArray &&
-                        !SymbolList.ValueTypes.Contains(item.AsTypeName) &&
-                        (item.AsTypeDeclaration == null || (!ClassModuleDeclaration.HasDefaultMember(item.AsTypeDeclaration) &&
-                        item.AsTypeDeclaration.DeclarationType != DeclarationType.Enumeration &&
-                        item.AsTypeDeclaration.DeclarationType != DeclarationType.UserDefinedType)) &&
-                        (item.DeclarationType == DeclarationType.Variable ||
-                         item.DeclarationType == DeclarationType.Parameter));
-
-            var interestingMembers =
-                State.AllUserDeclarations.Where(item =>
-                    (item.DeclarationType == DeclarationType.Function || item.DeclarationType == DeclarationType.PropertyGet)
-                    && !item.IsArray
-                    && item.IsTypeSpecified
-                    && !SymbolList.ValueTypes.Contains(item.AsTypeName) 
-                    && (item.AsTypeDeclaration == null // null if unresolved (e.g. in unit tests)
-                        || (item.AsTypeDeclaration.DeclarationType != DeclarationType.Enumeration && item.AsTypeDeclaration.DeclarationType != DeclarationType.UserDefinedType 
-                            && item.AsTypeDeclaration != null 
-                            && !ClassModuleDeclaration.HasDefaultMember(item.AsTypeDeclaration))));
-
-            var interestingReferences = interestingDeclarations
-                    .Union(interestingMembers.SelectMany(item =>
-                        item.References.Where(reference => reference.ParentScoping.Equals(item) && reference.IsAssignment)
-                    .Select(reference => reference.Declaration)))
-                    .SelectMany(declaration =>
-                        declaration.References.Where(reference =>
-                        {
-                            var letStmtContext = ParserRuleContextHelper.GetParent<VBAParser.LetStmtContext>(reference.Context);
-                            return reference.IsAssignment && letStmtContext != null && letStmtContext.LET() == null;
-                        })
-                    );
-
-
-            return interestingReferences.Select(reference => new ObjectVariableNotSetInspectionResult(this, reference));
+            var allInterestingDeclarations =
+                VariableRequiresSetAssignmentEvaluator.GetDeclarationsPotentiallyRequiringSetAssignment(State.AllUserDeclarations);
+
+            var candidateReferencesRequiringSetAssignment = 
+                allInterestingDeclarations.SelectMany(dec => dec.References)
+                    .Where(dec => !IsIgnoringInspectionResultFor(dec, AnnotationName))
+                        .Where(reference => reference.IsAssignment);
```

Comment:
Please align this ".Where" with the previous--otherwise, it looks like you are `.Where`ing a collection nested in the first `.Where` call.